patched-modules/octicons.js

```diff
@@ -0,0 +1,81 @@
+// NOTE: Patched version of octicons.
+// Changes:       Added else check what returns default values if
+//                provided value undefined, null or something else,
+//                what can not be in options.
+// Upstream Bugs: If provided empty values into class options
+//                like string (eta '' or ``) this shows empty
+//                class without additional context.
+//                Return `class=""`, should be 'class="octicon octicon-alert"'
+
+const data = require('octicons/build/data.json')
+const objectAssign = require('object-assign')
+
+for (const key of Object.keys(data)) {
+  // Returns a string representation of html attributes
+  const htmlAttributes = (icon, options) => {
+    const attributes = []
+    const attrObj = objectAssign({}, data[key].options, options)
+
+    // If the user passed in options
+    if (options) {
+      // If any of the width or height is passed in
+      if (options['width'] || options['height']) {
+        attrObj['width'] = options['width']
+          ? options['width']
+          : (parseInt(options['height']) * data[key].options['width']) / data[key].options['height']
+        attrObj['height'] = options['height']
+          ? options['height']
+          : (parseInt(options['width']) * data[key].options['height']) / data[key].options['width']
+      } else {
+        attrObj['height'] = data[key].width
+        attrObj['width'] = data[key].height
```

Comment:
It looks like `options['width']`, `options['height']`, `options['class']`, etc. are set in the "Set all the default options" section below -- are these 'else' clauses needed?

And if so, is there any alternative to patching? For example wrap our use of `.toSVG()` with something that fills in the defaults at that point in our own files? It will probably cause us less work over time (e.g. when octicons_node has new releases) if we don't use a patched version.

Reply:
5e9f9562d550fd3b1af74d08a520bc11e437e8e6 adds hack to delete `null`'s options for `.toSVG()`.